modules/evm/src/runner/stack.rs

```diff
@@ -678,11 +678,7 @@ impl<'vicinity, 'config, T: Config> BackendT for SubstrateStackState<'vicinity,
 	}
 
 	fn original_storage(&self, address: H160, index: H256) -> Option<H256> {
-		if let Some(value) = self.substate.known_original_storage(address, index) {
-			Some(value)
-		} else {
-			Some(self.storage(address, index))
-		}
+		self.substate.known_original_storage(address, index)
 	}
```

Comment:
Any particular reason for this? Would change the behavior, although I do find it a bit wierd that this function returns an `Option<H256>` when it has no branch that could return `None`

Reply:
That’s generated by cargo clippy, so this might be equivalent. If not, I am happy to revert those unrelated changes.

Reply:
Yeah the change seems reasonable, but will change the behavior to return `None` rather then `Some(H256::default())`. Maybe jingbo will have insight into where this function is used because I can't quite figure that out

Reply:
This looks like a false positive from cargo clippy. The new code is clearly not same as the old one. Please revert.

Reply:
What's your rust version? I use `nightly-2022-07-14` but can't reproduce the Clippy warning.

Reply:
I was on `rustc 1.62.0-nightly (8f36334ca 2022-04-06)`. Anyway, gonna revert those clippy changes.